### ISSUE TITLE
Sending an SMS hides the fact multiple messages were sent

### ISSUE THREAD

**Issue by dragonmantank** [Nexmo/nexmo-php#221]
From discussion on Discord with a user:

> I am working with SMS Message price, now SMS can consist of more than 1 message parts, right? concatenation and stuff. Nexmo charges for each SMS part.
>
> If we look at this getPrice method in nexmo-php , does it return the cost of Message as a whole? or just one part (last one)?

Digging into this more, the `Nexmo\Message\Message` object assumes a single message is returned by the API. In the case where multiple messages are returned, such as a long message that is split in two, it only returns data for the last chunk. The messages send properly, the SDK just doesn't report back data for the _other_ chunk.

The SDK should not lose the context of multiple messages like that, but this will be a breaking change to fix. 

Short-term fix is to use the raw request data:

```php
$sms = $nexmo->message->send(...);
$response = $sms->getResponseData();
$total = 0.0;
foreach ($response['messages'] as $message) {
    $total += (float)$message['message-price'];
}
```

**Comment by awebartisan** [comment]
Maybe we can consider `BCMath` to perform the addition and not loose precision? 

- This extension does not come pre-packaged with PHP but can be installed easily.
- The Windows version of PHP has built-in support for this extension.

**Comment by dragonmantank** [comment]
This is partially fixed in v2.2.0 in the new `sms()` client (which isn't available in Laravel yet, which is where the original problem was discovered).

The above suggestion is basically the same as the current solution, just the returned data from the API is more apparent that it is iterable.

```php
$response = $client->sms()->send(new SMS('15551111234', '15552227894', 'Test message'));
$total = 0.0;
foreach ($response as $message) {
    $total += (float) $message->getMessagePrice();
}
```

As for using the BC Math extension, the end user is free to implement this to keep float precision. I did not add a way to get the total cost because of this, but the new interface is much more up-front that the result from sending an SMS is an iterable object they can pull individual message details out of. 

I'm closing this, but leaving #219 open until the Laravel package is updated to support the newer `sms()` client, which will be a backwards-compatibility-breaking change.
